pipeline-to-taskrun/pkg/reconciler/pipelinetotaskrun/taskrun.go

```diff
@@ -37,6 +37,25 @@ func getMergedTaskRun(run *v1alpha1.Run, pSpec *v1beta1.PipelineSpec, taskSpecs
 		})
 	}
 
+	// if an optional workspace isn't provided, we don't need to remap it but we still need to declare it
+	// in order for any variable interpolation to work
+	for pt, mappings := range newWorkspaceMapping {
+		for oldName, newName := range mappings {
+			if newName == "" {
+				taskSpec := taskSpecs[pt]
+				for _, ws := range taskSpec.Workspaces {
+					if ws.Name == oldName {
+						tr.Spec.TaskSpec.Workspaces = append(tr.Spec.TaskSpec.Workspaces, v1beta1.WorkspaceDeclaration{
+							Name:        oldName,
+							Description: ws.Description,
+							Optional:    ws.Optional,
+						})
+					}
+				}
+			}
+		}
+	}
```

Comment:
I see what you mean! I had a really hard time understanding what was going on in the original version XD

I've made a slightly different change - instead of using `getNewWorkspaceMapping` to both determine the new mapping and to detect optional workspaces (and using "" to communicate that), I moved the optional workspace logic into a new function `getUnboundOptionalWorkspaces` that returns the optional workspaces.

let me know what you think - if this is better, and/or if you still prefer the above version

Reply:
💯  This new version is much more readable! Thanks for making the change!

Reply:
np thanks for the feedback @wlynch !!